Makefile

```diff
@@ -410,9 +410,15 @@ include prebuilds.mk
 -include $(filter %.d,$(STDOBJS:%.o=%.d))
 -include $(filter %.d,$(HVLOGOBJS:%.o=%.d))
 
+# 'make all' calculates the current checksum of all .h and .hpp files, storing the checksum in a file. Then it decides whether to run 'make clean' or 'make standard' based on whether any .h and .hpp files have been altered
+HEADER_CHECKSUM_FILE=.header_checksum
+
 all:
-	@$(MAKE) clean
-	@$(MAKE) standard
```

Comment:
I still think a simple `build || { clean && build }` should work too.

Reply:
I'm not sure what that means, maybe you could elaborate.

YM suggested the original functionality of `make all` be replaced with this PR. So you'll have have to talk to him if you want to preserve the original `make all`.

Reply:
`@$(MAKE) standard || { @$(MAKE) clean && @$(MAKE) standard }`

Reply:
When you don't run `make clean` after you edit the header files, that doesn't cause `make standard` to fail, it compiles successfully, but it'll have bugs and cause issues.

Reply:
Ah I see.